src/packagedef.jl

```diff
@@ -874,6 +882,33 @@ function revise(; throw=false)
                 queue_errors[(pkgdata, file)] = (err, catch_backtrace())
             end
         end
+        # Handle binding invalidations
+        if !isempty(reeval_methods)
+            handled = typeof(reeval_methods)()
+            while !isempty(reeval_methods)
+                m = pop!(reeval_methods)
+                methinfo = get(CodeTracking.method_info, m.sig, missing)
+                methinfo === missing && continue
+                if length(methinfo) != 1
+                    @warn "Multiple definitions for $(m.sig) found, skipping reevaluation"
+                    continue
+                end
+                Base.delete_method(m)  # ensure that "old data" doesn't get run with "old methods"
+                _, ex = methinfo[1]
+                eval_with_signatures(m.module, ex; mode=:eval)
```

Comment:
Note to self: I was surprised that dependent types (e.g., `PointWrapper`) don't experience binding invalidation until their constructors are re-evaluated. `Core.eval(m.module, ex)` triggers the binding invalidation and gets us farther in the tests, but it doesn't add the appropriate signatures to `CodeTracking.method_info`. My sense is that JuliaInterpreter/JuliaLowering need to trigger binding invalidation from `LoweredCodeUtils.methoddef!`.